core/src/main/java/io/grpc/internal/GzipInflatingBuffer.java

```diff
@@ -190,7 +192,7 @@ boolean isStalled() {
    */
   boolean hasPartialData() {
     checkState(!closed, "GzipInflatingBuffer is closed");
-    return gzippedData.readableBytes() != 0 || (inflaterInputStart < inflaterInputEnd);
+    return gzipMetadataReader.readableBytes() != 0;
```

Comment:
`&& state == HEADER`

Reply:
Done (`|| state != HEADER`), also added test case to cover this.